### PR TITLE
Feat/allow undoing service cancellations

### PR DESCRIPTION
**Pull request by FredericEspiau**
## Allow undoing service cancellations

Reference: SCFRCA-1244

### Description of the Change

Displays a banner so users can undo cancellations more easily

### CODE REVIEW

File: client/app/account/billing/autoRenew/billing-autoRenew.controller.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
  * Autorenew services configuration
  */
 angular.module("Billing.controllers").controller("Billing.controllers.AutoRenew", [
+    "$rootScope",
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/billing-autoRenew.controller.js]: I won't refactor lines that I haven't touched on this PR, no need to ask

File: client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.controller.js [hunk 2]
```diff
@@ -1,56 +1,68 @@
...
-        const AUTO_RENEW_TYPES = ["automaticV2014", "automaticV2016", "automaticForcedProduct"];
-        const renewType = $scope.selectedServices[0].service ? $scope.selectedServices[0].service.renewalType : $scope.selectedServices[0].renewalType;
-        if ($scope.selectedServices[0].renew.automatic && !AUTO_RENEW_TYPES.includes(renewType)) {
-            setManual()
-                .then(setDelete)
-                .catch(errorFunction)
-                .finally(() => {
-                    $rootScope.$broadcast(BillingAutoRenew.events.AUTORENEW_CHANGED);
-                    $scope.resetAction();
+angular
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.controller.js]: I did refactor this file though

File: client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.html [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
-            <p data-ng-bind-html="'autorenew_service_delete_confirm' | translate:{
-                   t0: selectedServices[0].domain,
-                   t1: selectedServices[0].expirationText
-               }">
-            </p>
-            <p data-ng-bind-html="'autorenew_service_delete_warn' | translate "></p>
-        </div>
-
-    </div>
+<div data-ng-controller="billingAutoRenewDeleteCtrl as $ctrl">
```
Comment (FredericEspiau) [inline client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.html]: This one too

File: client/app/account/billing/autoRenew/hosting/actions-menu.html [hunk 4]
```diff
@@ -34,7 +34,7 @@
                 class="btn btn-link"
                 data-ng-class="{ 'disabled' : !canResiliate(service.subProducts.hosting_web, user) }"
                 data-ng-disabled="!canResiliate(service.subProducts.hosting_web, user)"
-                data-ng-click="canResiliate(service.subProducts.hosting_web, user) && resiliateService(service.subProducts.hosting_web)">
+                data-ng-click="resiliateService(service.subProducts.hosting_web)">
```
Comment (antleblanc) [inline client/app/account/billing/autoRenew/hosting/actions-menu.html]: Some refacto are necessary here:

```diff
     <li data-ng-if="!service.subProducts.hosting_web.renew.deleteAtExpiration">
         <button type="button"
-                class="btn btn-link"
                 data-ng-class="{ 'disabled' : !canResiliate(service.subProducts.hosting_web, user) }"
                 data-ng-disabled="!canResiliate(service.subProducts.hosting_web, user)"
-                data-ng-click="resiliateService(service.subProducts.hosting_web)">
-            <i class="fa fa-trash" aria-hidden="true"></i>
-            <span data-translate="autorenew_service_resiliate_button"></span>
+                data-ng-click="resiliateService(service.subProducts.hosting_web)"
+                data-translate="autorenew_service_resiliate_button">
         </button>
     </li>
```

Would be great to spread this to all actions (`<li>`).

File: client/app/account/billing/autoRenew/common/actions-menu.html [hunk 5]
```diff
@@ -41,7 +41,7 @@
                 class="btn btn-link"
                 data-ng-class="{ 'disabled' : !canResiliate(service, user) }"
                 data-ng-disabled="!canResiliate(service, user)"
-                data-ng-click="canResiliate(service, user) && resiliateService(service)"
```
Comment (frenautvh) [inline client/app/account/billing/autoRenew/common/actions-menu.html]: in case it's not used anymore, you can remove canResiliate method

Comment (FredericEspiau) [reply client/app/account/billing/autoRenew/common/actions-menu.html]: Well... read again 😄

File: client/app/account/billing/autoRenew/billing-autoRenew.controller.js (hunk 1 above)
Comment (antleblanc) [reply client/app/account/billing/autoRenew/billing-autoRenew.controller.js]: What's the reason of such a comment?

File: client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.controller.js [hunk 6]
```diff
@@ -1,56 +1,68 @@
...
+                    this.$scope.$emit(this.AUTORENEW_EVENT.TERMINATE_AT_EXPIRATION, formattedServices);
                 });
-        } else {
-            setDelete()
-                .catch(errorFunction)
+        }
+
+        switchRenewalModeToManualIfNeeded () {
+            const AUTO_RENEW_TYPES = ["automaticV2014", "automaticV2016", "automaticForcedProduct"];
+            const renewType = _(this.serviceToDisplay).get("service.renewalType") || this.serviceToDisplay.renewalType;
```
Comment (marie-j) [inline client/app/account/billing/autoRenew/delete/billing-autoRenew-delete.controller.js]: Why not write directly `_(this.serviceToDisplay).get("service.renewalType",this.serviceToDisplay.renewalType)` instead ? 😄

**Comment by jleveugle** [comment]
Hello @FredericEspiau,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**